File: university-ios-assignment-5/AboutViewController.m
//
//  AboutViewController.m
//  university-ios-assignment-5
//
//  Created by admin on 14/07/16.
//  Copyright © 2016 tanya. All rights reserved.
//

#import "AboutViewController.h"
#import "DetailInfoViewController.h"

NSString *kSegueidentifierShowDetail = @"showDetail";

@interface AboutViewController () <DetailInfoViewControllerDelegate>
@property (weak, nonatomic) IBOutlet UITextView *textView;
@property (weak, nonatomic) IBOutlet UIScrollView *scrollView;
@property (weak, nonatomic) IBOutlet UIView *firstButton;
@property (weak, nonatomic) IBOutlet UIButton *secondButton;
@property (weak, nonatomic) IBOutlet UIButton *thirdButton;

@end

@implementation AboutViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}


- (IBAction)firstButtonTouchedUpInside:(id)sender {
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Alert title" message:@"Alert message" preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction* ok = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil];
    [alertController addAction:ok];
    
    [self presentViewController:alertController animated:YES completion:nil];
}
- (IBAction)secondButtonTouchedUpInside:(id)sender {
    UIAlertController * alert = [UIAlertController
                                 alertControllerWithTitle:@"Show detail?"
                                 message:@"Do you want to look at detail?"
                                 preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction* yesButton = [UIAlertAction
                                actionWithTitle:@"Yes, please"
                                style:UIAlertActionStyleDefault
                                handler:^(UIAlertAction * action) {
                                    [self performSegueWithIdentifier:kSegueidentifierShowDetail sender:self];
                                }];
    
    UIAlertAction* noButton = [UIAlertAction
                               actionWithTitle:@"No, thanks"
                               style:UIAlertActionStyleDefault
                               handler:^(UIAlertAction * action) {
                                   
                               }];
    
    [alert addAction:yesButton];
    [alert addAction:noButton];
    
    [self presentViewController:alert animated:YES completion:nil];
}

- (void)closeDetailInfo
{
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([segue.identifier isEqualToString:kSegueidentifierShowDetail])
    {
        UINavigationController *navigationController = segue.destinationViewController;
        DetailInfoViewController *playerDetailsViewController = navigationController.viewControllers.lastObject;
        playerDetailsViewController.delegate = self;
    }
}

@end

File: Task_5/Task_5/AboutViewController.swift
//
//  AboutViewController.swift
//  Task_5
//
//  Created by Kirill Asyamolov on 11/11/16.
//  Copyright © 2016 Kirill Asyamolov. All rights reserved.
//

import UIKit

class AboutViewController: UIViewController, DetailsViewControllerDelegate {

    let segueIdentifierShowDetails = "showDetails"
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    func closeDetailsPopup() {
        self.dismiss(animated: true, completion: nil)
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        print(segue.identifier)
        if segue.identifier == segueIdentifierShowDetails {
            let uiViewController = segue.destination
            let detailsViewController: DetailsViewController = uiViewController.childViewControllers.last as! DetailsViewController
            detailsViewController.delegate = self

        }
    }
    
    @IBAction func showAlert_clickHandler(_ sender: AnyObject) {
        let alertController = UIAlertController(title: "Title", message: "Test message", preferredStyle: .alert)
        let okAction = UIAlertAction(title: "OK", style: .default)
        alertController.addAction(okAction)
        
        present(alertController, animated: true, completion: nil)
        
    }
    
    @IBAction func showAnotherAlert_clickHandler(_ sender: AnyObject) {
        let alertController = UIAlertController(title: "Title", message: "Show detailed info?", preferredStyle: .alert)
        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
        alertController.addAction(cancelAction)
        let okAction = UIAlertAction(title: "OK", style: .default) { (action) in
            self.performSegue(withIdentifier: self.segueIdentifierShowDetails, sender: self)
        }
        alertController.addAction(okAction)
        present(alertController, animated: true, completion: nil)
    }
    

}
